### ISSUE TITLE
Stroke gradient not supported?

### ISSUE THREAD

**Issue by gbuela** [swhitty/SwiftDraw#17]
Hello, I'm trying to set some shape's stroke to a gradient and it's not getting rendered. Is that not supported? Is it something that you could add?

**Comment by swhitty** [comment]
HI @gbuela do you have an SVG sample file I can check.  

There is basic support for stroking with a gradient on Apple platforms.

See this sample:
https://github.com/swhitty/SwiftDraw/blob/main/Samples/angry.svg

**Comment by gbuela** [comment]
Here's a sample

```
<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN" "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    xmlns:krita="http://krita.org/namespaces/svg/krita"
    xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
    width="150"
    height="150">
<defs>
  <linearGradient id="grad1" x1="0" y1="0" x2="0" y2="1" spreadMethod="pad">
    <stop stop-color="#ff0000" offset="0%" stop-opacity="1"/>
    <stop stop-color="#ffff00" offset="34%" stop-opacity="1"/>
    <stop stop-color="#00ff00" offset="50%" stop-opacity="1"/>
    <stop stop-color="#00ffff" offset="66%" stop-opacity="1"/>
    <stop stop-color="#0000ff" offset="100%" stop-opacity="1"/>
  </linearGradient>
</defs>
<circle cx="50" cy="50" r="40" stroke="url(#grad1)" stroke-width="10" fill="none"/>
</svg>
```

**Comment by swhitty** [comment]
I have extended the logic for determining the gradient start, end points to shapes and polygons.  Try it out on the `main` branch
<img width="395" alt="Screen Shot 2022-12-23 at 6 42 46 am" src="https://user-images.githubusercontent.com/867662/209213783-b4a0118d-e29c-47e1-9f93-1c2911e63d0b.png">

**Comment by gbuela** [comment]
Yup it seems to work fine!

**Comment by swhitty** [comment]
Included in [0.13.3](https://github.com/swhitty/SwiftDraw/releases/tag/0.13.3) release
